File: LeetCode June Challenge/Triangle.java
import java.util.*;
class Solution {
    public int minimumTotal(List<List<Integer>> triangle) {
        int n = triangle.size();
        int[] dp = new int[n];
        Arrays.fill(dp, 0);
        for(int i = 0; i < n; ++i){
            dp[i] = triangle.get(n - 1).get(i);
        }
        for(int i = n - 2; i >= 0; --i){
            for(int j = 0; j < triangle.get(i).size(); ++j){
                dp[j] = Math.min(dp[j], dp[j + 1]) + triangle.get(i).get(j);
            }
        }
        return dp[0];
    }
}
public class Triangle {
    public static void main(String[] args) {
        List<List<Integer>> triangle = new ArrayList<>();
        Solution s = new Solution();
        System.out.println(s.minimumTotal(triangle));
    }
}

File: Year-2022/LeetCode June Challenge/triangle.js
var minimumTotal = function(triangle) {
    let n = triangle.length;
    let dp = new Array(n);
    dp.fill(0);
    for(let i = 0; i < n; ++i){
        dp[i] = triangle[n - 1][i];
    }
    for(let i = n - 2; i >= 0; --i){
        for(let j = 0; j < triangle[i].length; ++j){
            dp[j] = Math.min(dp[j], dp[j + 1]) + triangle[i][j];
        }
    }
    return dp[0];
};
